packages/opentelemetry-api/src/diag/logLevel.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DiagAPI } from '../api/diag';
+import { DiagLogger, DiagLogFunction } from './logger';
+
+/**
+ * Defines the available internal logging levels for the diagnostic logger, the numeric values
+ * of the levels are defined to match the original values from the initial LogLevel to avoid
+ * compatibility/migration issues for any implementation that assume the numeric ordering.
+ */
+export enum DiagLogLevel {
+  /** DIagnostic Logging level setting to disable all logging (except and forced logs) */
+  NONE = -99,
+
+  /** Identifies a terminal situation that would cause the API to completely fail to initialize,
+   * if this type of error is logged functionality of the API is not expected to be functional.
+   */
+  TERMINAL = -2,
+
+  /** Identifies a critical error that needs to be addressed, functionality of the component
+   * that emits this log detail may non-functional.
+   */
+  CRITICAL = -1,
+
+  /** Identifies an error scenario */
+  ERROR = 0,
+
+  /** Identifies a warning scenario */
+  WARN = 1,
+
+  /** General informational log message */
+  INFO = 2,
+
+  /** General debug log message */
+  DEBUG = 3,
+
+  /** Detailed trace level logging should only be used for development, should only be set
```

Comment:
Not sure I understand why? -- I did see bart's comment on renaming to VERBOSE which makes a lot of sense to avoid overloading "Trace" yet again.

Reply:
To avoid letting it get checked in. If it fails lint it will fail CI

Reply:
I think I understand what your talking about, I don't think we want to block this level of message being added... The comment is that when being "used" it is not recommended that they deploy to a production system with this setting as it would (possible) cause a bucket load of internal logging...